src/ResourceManager/Sql/Commands.Sql/help/Add-AzureRmSqlElasticJobStep.md

```diff
@@ -0,0 +1,495 @@
+```
+
+### -CredentialName
+The credential name
+
+```yaml
+Type: String
+Parameter Sets: (All)
+Aliases:
+
+Required: True
+Position: 6
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -DefaultProfile
+The credentials, account, tenant, and subscription used for communication with Azure.
+
+```yaml
+Type: IAzureContextContainer
+Parameter Sets: (All)
+Aliases: AzureRmContext, AzureCredential
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -InitialRetryIntervalSeconds
+The initial retry interval seconds
+
+```yaml
+Type: Int32
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -JobName
+The job name
+
+```yaml
+Type: String
+Parameter Sets: Default Parameter Set, Add Job Step With Output Database Object Parameter Set, Add Job Step With Output Database Resource Id Parameter Set
+Aliases:
+
+Required: True
+Position: 3
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -JobObject
+The job object
+
+```yaml
+Type: AzureSqlElasticJobModel
+Parameter Sets: Input Object Parameter Set, Add Job Step With Output Database Object Parameter Set Using Job Object, Add Job Step With Output Database Id Parameter Set Using Job Object
+Aliases:
+
+Required: True
+Position: 0
+Default value: None
+Accept pipeline input: True (ByValue)
+Accept wildcard characters: False
+```
+
+### -JobResourceId
+The job resource id
+
+```yaml
+Type: String
+Parameter Sets: Resource Id Parameter Set, Add Job Step With Output Database Object Parameter Set Using Job Resource Id, Add Job Step With Output Database Id Parameter Set Using Job Resource Id
+Aliases:
+
+Required: True
+Position: 0
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -MaximumRetryIntervalSeconds
+The maximum retry interval seconds
+
+```yaml
+Type: Int32
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Name
+The job step name
+
+```yaml
+Type: String
+Parameter Sets: (All)
+Aliases: StepName
+
+Required: True
+Position: 4
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -OutputCredentialName
+The output credential name
+
+```yaml
+Type: String
+Parameter Sets: Add Job Step With Output Database Object Parameter Set, Add Job Step With Output Database Resource Id Parameter Set, Add Job Step With Output Database Object Parameter Set Using Job Object, Add Job Step With Output Database Id Parameter Set Using Job Object, Add Job Step With Output Database Object Parameter Set Using Job Resource Id, Add Job Step With Output Database Id Parameter Set Using Job Resource Id
```

Comment:
These param set names are verbose which makes them hard to read. Could you shorten them, e.g. `WithOutput`, `WithOutputId`, `WithOutputObject`, etc ?